Add App-like PWA article

tomayac opened the pull request:
Fixes #1441

Changes proposed in this pull request:

- Add the article

inexorabletash commented:
_"being standardized at the moment"_ - please be careful about the "proposed", "specified" and "standardized".  We should reserve "standard" for when there is adoption browsers and/or adoption into a standards body. Just because something has an explainer or spec doesn't mean it's a standard.

tomayac commented:
@tomayac Maybe need to use device frames for the mobile screenshots. (via @mrdoob).

dalmaer commented:
FYI: "Push notifications have become a bit of an annoyance on the web". May be a good idea to point to the Reparo blog post that is about to go out that talks about how we are attacking the annoyance factor.

tomayac commented:
Add a warning against overuse of `user-select:none` (via @jyasskin).

tomayac commented:
> _"being standardized at the moment"_ - please be careful about the "proposed", "specified" and "standardized". We should reserve "standard" for when there is adoption browsers and/or adoption into a standards body. Just because something has an explainer or spec doesn't mean it's a standard.

I have [reworded](https://github.com/GoogleChrome/web.dev/pull/2731/commits/dfd57e416bad4a9853f78a92f6eb87bd9686649f) this now.

tomayac commented:
> FYI: "Push notifications have become a bit of an annoyance on the web". May be a good idea to point to the Reparo blog post that is about to go out that talks about how we are attacking the annoyance factor.

Added in [this commit](https://github.com/GoogleChrome/web.dev/pull/2731/commits/dfd57e416bad4a9853f78a92f6eb87bd9686649f).

tomayac commented:
> Add a warning against overuse of `user-select:none` (via @jyasskin).

Addressed in [this commit](https://github.com/GoogleChrome/web.dev/pull/2731/commits/b8814b577528f0c9bf671ef1763c6efcb0bac0ff).

tomayac commented:
The hidden section revealing that custom URL schemes aren't supported was disappointing. You might want to consider a table at the end so that people can easily see the comparison.

(via @adewale)

tomayac commented:
> @tomayac Maybe need to use device frames for the mobile screenshots. (via @mrdoob).

Addressed in [this commit](https://github.com/GoogleChrome/web.dev/pull/2731/commits/951bad48160f0750853e33f6203122ac01437388).

tomayac commented:
> The hidden section revealing that custom URL schemes aren't supported was disappointing. You might want to consider a table at the end so that people can easily see the comparison.
> 
> (via @adewale)

Addressed in [this commit](https://github.com/GoogleChrome/web.dev/pull/2731/commits/c873c19c1fbfed4435f5a2f5ca1a7f558af3b96c).

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+In-app animations are snappy and smooth in Podcasts. For example, when I open the "Episode Notes" drawer on the right, it elegantly slides in. When I remove one episode from my downloads, the remaining episodes float up and consume the screen real estate that was freed by the deleted episode.
+
+<figure class="w-figure">
+  <img src="./image14.png" alt="The Podcasts app with the 'Episode Notes' drawer expanded." width="600">
```
```suggestion
  <img src="./image14.png" class="w-screenshot w-screenshot--filled" alt="The Podcasts app with the 'Episode Notes' drawer expanded.">
```

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+
+## Snappy animations
+
+In-app animations are snappy and smooth in Podcasts. For example, when I open the "Episode Notes" drawer on the right, it elegantly slides in. When I remove one episode from my downloads, the remaining episodes float up and consume the screen real estate that was freed by the deleted episode.
```
```suggestion
In-app animations are snappy and smooth in Podcasts. For example, when I open the **Episode Notes** drawer on the right, it elegantly slides in. When I remove one episode from my downloads, the remaining episodes float up and consume the screen real estate that was freed by the deleted episode.
```

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+The Podcasts app on iOS can surface content in other locations than the actual application, for example, in the system's Widgets view, or in the form of a Siri Suggestion. Having proactive, usage-based calls-to-action that just require a tap to interact with can greatly increase the re-engagement rate of an app like Podcasts.
+
+<figure class="w-figure">
+  <img src="./image6.png" alt="iOS Widget view showing the Podcasts app suggesting a new episode of a podcast." width="300">
```
```suggestion
  <img src="./image6.png" class="w-screenshot w-screenshot--filled" alt="iOS Widget view showing the Podcasts app suggesting a new episode of a podcast." width="300">
```

kaycebasques replied:
This image is quite busy. I suggest setting a plain black background and removing the **TIPS** and **STOCKS** notifications.

tomayac replied:
I want it to be realistic, so not just one widget. I have added a red highlight box now.

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+  By marking up interesting content in your app as suitable for <a href="https://developers.google.com/search/docs/data-types/speakable">speakable</a>
+  audio playback and by using <a href="https://developers.google.com/search/docs/guides/search-gallery">structured markup</a> in general,
+  you can help search engines and virtual assistants like the Google Assistant present your offerings in an ideal light.
+{% endDetails %}
```
Can I get my PWA notification to show up on my iOS device? This seems misleading to me.

tomayac replied:
Conditions apply, yes. Let me find a way to address the browser compatibility questions.

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+{% endDetailsSummary %}
+  The <a href="https://web.dev/media-session/">Media Session API</a> lets you specify metadata like artwork, track titles, etc.
+  that then gets displayed on the lock screen, smartwatches, or other media widgets in the browser.
+{% endDetails %}
```
This section also implies to me that this feature is available on iOS

tomayac replied:
Same as before, browser compatibility is an open question.

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+When a podcast episode is playing, the Podcasts app shows a beautiful control widget on the lock screen that features metadata like the episode artwork, the episode title, and the podcast name.
+
+<figure class="w-figure">
+  <img src="./image17.png" alt="iOS media playback widget on the lock screen showing a podcast episode with rich metadata." width="300">
```
```suggestion
  <img src="./image17.png" class="w-screenshot w-screenshot--filled" alt="iOS media playback widget on the lock screen showing a podcast episode with rich metadata." width="300">
```

tomayac replied:
This uses a device frame now, no screenshot class needed.

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+For example, the iOS Podcasts app can optionally notify me of new episodes of podcasts I am subscribed to or recommend me new ones, as well as alert me of new app features.
+
+<figure class="w-figure">
+  <img src="./image12.png" alt="iOS Podcasts app in the 'Notifications' settings screen showing the 'New Episodes' notifications toggle activated." width="300">
```
```suggestion
  <img src="./image12.png" class="w-screenshot w-screenshot--filled" alt="iOS Podcasts app in the 'Notifications' settings screen showing the 'New Episodes' notifications toggle activated." width="300">
```

tomayac replied:
This uses a device frame, no screenshot class needed.

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+  allows your app to receive push notifications so you can notify your users about noteworthy events around your PWA.
+  For notifications that should fire at a known time in the future and that do not require a network connection,
+  you can use the <a href="https://web.dev/notification-triggers/">Notification Triggers API</a>.
+{% endDetails %}
```
This section also implies to me that this feature is available on iOS

tomayac replied:
Same as before.

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+  You can set app icon badges with the <a href="https://web.dev/badging-api/">Badging API</a>.
+  This is especially useful when your PWA has some notion of "unread" items or when you need a means
+  to unobtrusively draw the user's attention back to the app.
+{% endDetails %}
```
This section also implies to me that this feature is available on iOS

tomayac replied:
Same as before.

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+Whenever there are new episodes available for one of the podcasts I am subscribed to, an app icon badge on the Podcasts home screen icon appears, again encouraging me to re-engage with the app in a way that is not intrusive.
+
+<figure class="w-figure">
+  <img src="./image13.png" alt="iOS settings screen showing the 'Badges' toggle activated." width="310">
```
```suggestion
  <img src="./image13.png" class="w-screenshot w-screenshot--filled" alt="iOS settings screen showing the 'Badges' toggle activated." width="310">
```

tomayac replied:
Device frame.

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+Apps can optionally keep the screen awake, too, for example to display lyrics or captions.
+
+<figure class="w-figure">
+  <img src="./image21.png" alt="macOS Preferences in the 'Energy Saver' section." width="400">
```
```suggestion
  <img src="./image21.png" class="w-screenshot w-screenshot--filled" alt="macOS Preferences in the 'Energy Saver' section.">
```

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+  to unobtrusively draw the user's attention back to the app.
+{% endDetails %}
+
+## Media playback trumps energy saver settings
```
```suggestion
## Media playback takes precedence over energy saver settings
```
I know this is overly cautious and politically correct but we should avoid the word `trump` unless absolutely needed, in this case I don't think it is. Using the word is not worth the potential backlash. High risk, low benefit.

tomayac replied:
LOL, yeah!

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+A quick search for "podcast", "podcasts", or "apple podcasts" immediately turns the app up in the App Store.
+
+<figure class="w-figure">
+  <img src="./image22.png" alt="iOS App Store search for 'podcasts' reveals the Podcasts app." width="300">
```
```suggestion
  <img src="./image22.png" class="w-screenshot w-screenshot--filled" alt="iOS App Store search for 'podcasts' reveals the Podcasts app." width="300">
```

tomayac replied:
I now use device frames, so no screenshot class needed.

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+## App discovery through an app store
+
+While the Podcasts app is part of the macOS desktop experience, on iOS it needs to be installed from the App Store.
+A quick search for "podcast", "podcasts", or "apple podcasts" immediately turns the app up in the App Store.
```
```suggestion
A quick search for `podcast`, `podcasts`, or `apple podcasts` immediately turns the app up in the App Store.
```

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+## Conclusion
+
+PWAs have come a long way since their introduction in 2015.
+In the context of [Project Fugu 🐡](/fugu-status), we are working on closing the last remaining gaps.
```
```suggestion
In the context of [Project Fugu 🐡](/fugu-status), Chrome is working on closing the last remaining gaps.
```

kaycebasques commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+PWAs have come a long way since their introduction in 2015.
+In the context of [Project Fugu 🐡](/fugu-status), we are working on closing the last remaining gaps.
+By following even only some of the pieces of advice in this article,
+you can reach that holy grail app-like native feeling and make your users forget
```
The Holy Grail is a mythical thing that has never actually been found or achieved. I suggest rephrasing this because it currently implies PWAs will never actually achieve par with "native" apps.

tomayac replied:
Good point. And while probably true (we'll never fully get there), it may not be the best way to phrase it.

kaycebasques commented:
P.S. @tomayac sorry for the 30+ **Commit Suggestion** prompts. As I was halfway through my review I realized that I would have a lot of suggestions and that it might be tedious to use the GitHub UI to resolve them all. LMK if you want us to use a different workflow in the future.

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+description: |
+  Learn how to make your Progressive Web App feel like a "real" app by understanding
+  how to implement native app patterns with web technologies.
+hero: hero.jpg
```
It's purely decorative. If pages on this path don't have a hero anyway, let me kill it :-) 

```suggestion
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+<figure class="w-figure">
+  <img src="./image16.png" alt="Podcasts app with a downloaded episode of a podcast playing." width="600">
+  <figcaption class="w-figcaption">Downloaded podcast episodes can be played even without network.</figcaption>
+</figure>
```
Good point. Uploaded a new version.

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+  On Android, you can use the <a href="https://web.dev/contact-picker/">Contact Picker API</a> to select entries from the device's contacts manager.
+  If you offer both a native app and a PWA, you can use the <a href="https://web.dev/get-installed-related-apps/">Get Installed Related Apps API</a>
+  to check if the native app is installed, in which case you do not need to encourage the user to install the PWA or accept web push notifications.
+{% endDetails %}
```
The app can provide its own context-menu. See excalidraw.com for an example. I acknowledge, though, that this isn't the native one.

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,728 @@
+  How to do this on the web
+{% endDetailsSummary %}
+  The <a href="https://web.dev/web-share/">Web Share API</a> and the <a href="https://web.dev/web-share-target/">Web Share Target API</a>
+  allow your app to share and receive texts, files, and links to and from other applications on the device.
```
```suggestion
  allow your app to share and receive texts, files, and links to and from other applications on the device.
  Although it is not yet possible for a web app to add menu items to the operating system's built-in right-click menu, there are lots of other ways to link to and from other apps on the device.
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+  How to do this on the web
+{% endDetailsSummary %}
+  Media keys are supported by the <a href="https://web.dev/media-session/">Media Session API</a>.
+  Like that, users can profit from the hardware media keys on their physical keyboards, headphones, or even control the web app
```
LOL, I had to look up ESL readers and wasn't aware of any meme. Anyway, I have rephrased slightly now. 

```suggestion
  Like that, users can make use of the hardware media keys on their physical keyboards, headphones, or even control the web app
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,729 @@
+A more subtle thing but that is self-evident for a native application like Podcasts: none of the text labels are selectable and all texts blend in with the system font of the machine. Also my choice of system color theme (dark mode) is respected.
+
+<figure class="w-figure">
+  <img src="./image2.png" alt="The Podcasts app in dark mode." width="350">
+  <img src="./image7.png" alt="The Podcasts app in light mode." width="350">
```
```suggestion
  <img src="./image2.png" alt="The Podcasts app in dark mode." class="w-screenshot w-screenshot--filled" width="350">
  <img src="./image7.png" alt="The Podcasts app in light mode." class="w-screenshot w-screenshot--filled" width="350">
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+  <img src="./image2.png" alt="The Podcasts app in dark mode." width="350">
+  <img src="./image7.png" alt="The Podcasts app in light mode." width="350">
+  <figcaption class="w-figcaption">Podcasts supports Dark Mode and uses the default system font.</figcaption>
+</figure>
```
Added the classes. Leaving the comment open regarding the question of the side-by-side screenshots.

tomayac replied:
Actually, no, the classes made the screenshots look weirdly positioned.

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,438 @@
+  You can (and should), however, specify the <a href="https://web.dev/add-manifest/#display"><code>display</code></a> and the
+  <a href="https://web.dev/add-manifest/#theme-color"><code>theme-color</code></a> properties of the Web App Manifest to
+  determine the look and feel of your application window and to decide which default browser controls—potentially none of them—should be shown.
+{% endDetails %}
```
It's not as easy as upvoting a bug for all vendors. I suggest we hold off from adding a specific CtA here, and, once https://github.com/GoogleChrome/web.dev/issues/2127 has landed, point to it.

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,729 @@
+PWAs have come a long way since their introduction in 2015.
+In the context of [Project Fugu 🐡](/fugu-status), Chrome is working on closing the last remaining gaps.
+By following even only some of the pieces of advice in this article,
+you can reach that holy grail app-like native feeling and make your users forget
```
```suggestion
you can piece by piece get closer to that app-like native feeling and make your users forget
```

tomayac commented:
> P.S. @tomayac sorry for the 30+ **Commit Suggestion** prompts. As I was halfway through my review I realized that I would have a lot of suggestions and that it might be tedious to use the GitHub UI to resolve them all. LMK if you want us to use a different workflow in the future.

@kaycebasques The exact opposite, these suggestions are *exactly* the way I love to collaborate. Easy to apply, easy to modify (I can just edit your comment), and easy to comment on if I have a different opinion. Thanks for the review!

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,728 @@
+The most common app actions, **Search** for new content and **Check for New Episodes**, are available right from the context menu of the app in the Dock. Via the **Options** menu, I can also decide to open the app at login time.
+
+<figure class="w-figure">
+  <img src="./image18.png" alt="Podcasts app icon context menu showing the 'Search' and 'Check for New Episodes' options." width="267">
```
```suggestion
  <img src="./image18.png" alt="Podcasts app icon context menu showing the 'Search' and 'Check for New Episodes' options." class="w-screenshot w-screenshot--filled" width="267">
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,729 @@
+When you look at the Podcasts app window, you notice that it does not have a classic integrated title bar and toolbar, like, for example, the Safari browser window, but a customized experience that looks like a sidebar docked to the main player window.
+
+<figure class="w-figure">
+  <img src="./image8.png" alt="The Safari browser's integrated tile bar and toolbar.">
```
```suggestion
  <img src="./image8.png" class="w-screenshot" alt="The Safari browser's integrated tile bar and toolbar.">
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,729 @@
+Other native applications and even websites or emails can integrate with the Podcasts app by leveraging the `podcasts://` URL scheme. If I follow a link like [`podcasts://podcasts.apple.com/podcast/the-css-podcast/id1042283903`](podcasts://podcasts.apple.com/podcast/the-css-podcast/id1042283903) while in the browser, I am brought right into the Podcasts app and can decide to subscribe or listen t ...
+
+<figure class="w-figure">
+  <img src="./image20.png" class="w-screenshot w-screenshot--filled" alt="The Chrome browser showing a confirmation dialog asking the user whether they want to open the Podcasts app.">
```
```suggestion
  <img src="./image20.png" alt="The Chrome browser showing a confirmation dialog asking the user whether they want to open the Podcasts app.">
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,729 @@
+You may not immediately think of it, but the Podcasts app naturally integrates with the native file system. When I download a podcast episode, on my laptop it is stored in `~/Library/Group Containers/243LU875E5.groups.com.apple.podcasts/Library/Cache`. Unlike, say `~/Documents`, this directory is of course not meant to be accessed directly by regular users, but it is there.
+
+<figure class="w-figure">
+  <img src="./image5.png" class="w-screenshot w-screenshot--filled" alt="The macOS Finder navigated to the Podcasts app's system directory.">
```
```suggestion
  <img src="./image5.png" alt="The macOS Finder navigated to the Podcasts app's system directory.">
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,729 @@
+A more subtle thing but that is self-evident for a native application like Podcasts: none of the text labels are selectable and all texts blend in with the system font of the machine. Also my choice of system color theme (dark mode) is respected.
+
+<figure class="w-figure">
+  <img src="./image2.png" alt="The Podcasts app in dark mode." class="w-screenshot w-screenshot--filled" width="350">
+  <img src="./image7.png" alt="The Podcasts app in light mode." class="w-screenshot w-screenshot--filled" width="350">
```
```suggestion
  <img src="./image2.png" alt="The Podcasts app in dark mode." width="350">
  <img src="./image7.png" alt="The Podcasts app in light mode." width="350">
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,729 @@
+In-app animations are snappy and smooth in Podcasts. For example, when I open the **Episode Notes** drawer on the right, it elegantly slides in. When I remove one episode from my downloads, the remaining episodes float up and consume the screen real estate that was freed by the deleted episode.
+
+<figure class="w-figure">
+  <img src="./image14.png" class="w-screenshot w-screenshot--filled" alt="The Podcasts app with the 'Episode Notes' drawer expanded.">
```
```suggestion
  <img src="./image14.png" alt="The Podcasts app with the 'Episode Notes' drawer expanded.">
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,729 @@
+Apps can optionally keep the screen awake, too, for example to display lyrics or captions.
+
+<figure class="w-figure">
+  <img src="./image21.png" class="w-screenshot w-screenshot--filled" alt="macOS Preferences in the 'Energy Saver' section.">
```
```suggestion
  <img src="./image21.png" alt="macOS Preferences in the 'Energy Saver' section.">
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,729 @@
+
+A more subtle thing but that is self-evident for a native application like Podcasts: none of the text labels are selectable and all texts blend in with the system font of the machine. Also my choice of system color theme (dark mode) is respected.
+
+<figure class="w-figure">
+  <img src="./image2.png" alt="The Podcasts app in dark mode." width="350">
+  <img src="./image7.png" alt="The Podcasts app in light mode." width="350">
+  <figcaption class="w-figcaption">Podcasts supports Dark Mode and uses the default system font.</figcaption>
+</figure>
```
```suggestion
<div class="w-columns">
<figure class="w-figure">
  <img src="./image2.png" alt="The Podcasts app in dark mode." width="350">
  <img src="./image7.png" alt="The Podcasts app in light mode." width="350">
  <figcaption class="w-figcaption">Podcasts supports Dark Mode and uses the default system font.</figcaption>
</figure>
</div>
```

Found it in https://web.dev/handbook/web-dev-components/#columns.

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,734 @@
+The Podcasts app integrates naturally with other applications. For example, when I right-click an episode that I like, I can share it to other apps on my device, like the Messages app. It also naturally integrates with the system clipboard. I can right-click any episode and copy a link to it.
+
+<figure class="w-figure">
+  <img src="./image19.png" alt="The Podcasts app's context menu invoked on a podcast episode with the 'Share Episode > Messages' option selected." width="600">
```
```suggestion
  <img src="./image19.png" class="w-screenshot" alt="The Podcasts app's context menu invoked on a podcast episode with the 'Share Episode > Messages' option selected." width="600">
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,734 @@
+The most common app actions, **Search** for new content and **Check for New Episodes**, are available right from the context menu of the app in the Dock. Via the **Options** menu, I can also decide to open the app at login time.
+
+<figure class="w-figure">
+  <img src="./image18.png" alt="Podcasts app icon context menu showing the 'Search' and 'Check for New Episodes' options." class="w-screenshot w-screenshot--filled" width="267">
```
```suggestion
  <img src="./image18.png" alt="Podcasts app icon context menu showing the 'Search' and 'Check for New Episodes' options." class="w-screenshot" width="267">
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,734 @@
+how your app is built (and why should they?), as long as it feels like a *real* app.
+
+## Acknowledgements
+
```
```suggestion

This article was reviewed by Kayce Basques, Joe Medley, Joshua Bell, Dion Almaer, Ade Oshineye,
Pete LePage, and Jeffrey Yasskin.
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,734 @@
+
+## Acknowledgements
+
+Hero image by [Leio McLaren](https://unsplash.com/@leio) on [Unsplash](https://unsplash.com/photos/R6IWobiTW1k).
```
```suggestion
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,734 @@
+  <img src="./image0.png" alt="An iPhone and a MacBook side by side, both running the Podcasts application." class="w-screenshot" width="600">
+  <figcaption class="w-figcaption">Apple Podcasts on iPhone and on macOS (<a href="https://support.apple.com/en-us/HT201859">Source</a>).</figcaption>
+</figure>
+
```
```suggestion

{% Aside 'caution' %}
Each native app feature that is listed below has a **How to do this on the web**
component that you can open for more details.
Please note that not all browsers on the various operating systems support all the APIs
that are listed. Be sure to carefully review the compatibility notes in the linked articles.
{% endAside %}
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,741 @@
+component that you can open for more details.
+Please note that not all browsers on the various operating systems support all the listed APIs
+or functionalities. Be sure to carefully review the compatibility notes in the linked articles.
+{% endAside %}
```
```suggestion
{% endAside %}

```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,742 @@
+## Conclusion
+
+PWAs have come a long way since their introduction in 2015.
+In the context of [Project Fugu 🐡](/fugu-status), Chrome is working on closing the last remaining gaps.
```
```suggestion
In the context of [Project Fugu 🐡](/fugu-status), the cross-company Chromium team is working on closing the last remaining gaps.
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,742 @@
+## Acknowledgements
+
+This article was reviewed by Kayce Basques, Joe Medley, Joshua Bell, Dion Almaer, Ade Oshineye,
+Pete LePage, and Jeffrey Yasskin.
```
```suggestion
Pete LePage, Sam Thorogood, and Jeffrey Yasskin.
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,742 @@
+  value for the <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/font-family"><code>font-family</code></a> CSS property allows you to
+  specify the default UI font of the system to be used for your app, whatever it may be in the concrete case.
+  Finally, your app can obey to the user's color scheme preference by respecting their <a href="https://web.dev/prefers-color-scheme/"><code>prefers-color-scheme</code></a> choice, with an optional <a href="https://github.com/GoogleChromeLabs/dark-mode-toggle">dark mode toggle</a>
+  to override it.
```
```suggestion
  to override it.
  Another thing to decide on might be what the browser should do when reaching the boundary of a scrolling area.
 This is possible with the <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/overscroll-behavior"><code>overscroll-behavior</code></a> CSS property.
```

tomayac commented:
@kaycebasques For the compatibility questions that you have raised, I don't want to make this article a maintenance nightmare with changing support situations requiring an update, so for now I have added a [⚠️ Caution aside](https://deploy-preview-2731--web-dev-staging.netlify.app/app-like-pwas/#content:~:text=Caution%3A%20Each%20native%20app%20feature%20that,compatibility%20notes%20in%20the%20linked%20articles.) that points people to the linked articles. 

I have also addressed all your other remarks. What do you think?

kaycebasques commented:
And @tomayac thank you for addressing my compat concerns

reillyeon commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,745 @@
+
+## Native file system integration
+
+You may not immediately think of it, but the Podcasts app naturally integrates with the native file system. When I download a podcast episode, on my laptop it is stored in `~/Library/Group Containers/243LU875E5.groups.com.apple.podcasts/Library/Cache`. Unlike, say `~/Documents`, this directory is of course not meant to be accessed directly by regular users, but it is there.
```
For an app like Podcasts is there a benefit to using the Native Filesystem API over IndexedDB? ~/Library isn't designed to be user-visible and is not accessible with the Native Filesystem API.

tomayac replied:
This is more to raise awareness that such an app uses the file system. The prose makes clear it's not meant to be user-visible. The section on [offline content available and media playable](https://deploy-preview-2731--web-dev-staging.netlify.app/app-like-pwas/#offline-content-available-and-media-playable) provides background on the different storage mechanisms and links to an article how to decide for one.

reillyeon replied:
My concern is that the text in this section might lead developers down the wrong path if they don't notice the other one. Can you link to that section from this one?

tomayac replied:
Added in https://github.com/GoogleChrome/web.dev/pull/2731/commits/5e22f4654bbe4a98fdd412fc7298fcba5f1e1ef2.

kaycebasques commented:
@tomayac we can also reduce the image sizes by making their width `1600px` maximum (the content column is `800px` so `1600px` keeps it looking crisp for 2x DPR screens)

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,745 @@
+## Acknowledgements
+
+This article was reviewed by Kayce Basques, Joe Medley, Joshua Bell, Dion Almaer, Ade Oshineye,
+Pete LePage, Sam Thorogood, and Jeffrey Yasskin.
```
```suggestion
Pete LePage, Sam Thorogood, Reilly Grant, and Jeffrey Yasskin.
```

tomayac commented:
> @tomayac please optimize all of the images

@kaycebasques done in https://github.com/GoogleChrome/web.dev/pull/2731/commits/49ee6fa965c1a911c1861d4c7fd8592e5dd4efff.

<img width="661" alt="Screen Shot 2020-05-25 at 14 48 52" src="https://user-images.githubusercontent.com/145676/82814399-63cf4600-9e97-11ea-8a5b-130cc6c5c9eb.png">

tomayac commented:
> @tomayac can you please also reduce the image sizes by making their width `1600px` maximum (the content column is `800px` so `1600px` keeps it looking crisp for 2x DPR screens)

@kaycebasques Done.

@robdodson I've used a [bash script from StackOverflow](https://stackoverflow.com/a/33599441/6255000) that others might find useful for similar tasks, or that we even might make part of the build step:

```bash
#!/bin/sh

set -e
maxwidth="1600"  # in pixels, the widest image you want to allow.

#find all .jpg in current dir and subdirectories
FILES="$(find . -iname '*.png')"

for imagefile in $FILES
do
if [ -f "$imagefile" ]; then
imgwidth=`sips --getProperty pixelWidth "$imagefile" | awk '/pixelWidth/ {print $2}'`
else
    echo "Oops, "$imagefile" does not exist." 
    exit
fi

if [ $imgwidth -gt $maxwidth ]; then
    echo " - Image too big. Resizing..."
    sips --resampleWidth $maxwidth "$imagefile" > /dev/null 2>&1  # to hide sips' ugly output 
    imgwidth=`sips --getProperty pixelWidth "$imagefile" | awk '/pixelWidth/ {print $2}'`
    imgheight=`sips --getProperty pixelHeight "$imagefile" | awk '/pixelHeight/ {print $2}'`
    echo " - Resized "$imagefile" to $imgwidth""px wide by $imgheight""px tall";
fi
done
```

kaycebasques commented:
FYI some of the images still seem bigger than necessary. But I'll just check out the branch and push up a commit myself.

![image](https://user-images.githubusercontent.com/4713486/82941882-fb8e6c00-9f4b-11ea-8f15-a2db3a823158.png)

kaycebasques commented on src/site/content/en/progressive-web-apps/progressive-web-apps.11tydata.js:
```diff
@@ -18,7 +18,7 @@ module.exports = {
       {
         title: 'Capable',
-        pathItems: [],
+        pathItems: ['app-like-pwas'],
```
@robdodson || @samthor please review the Percy change

kaycebasques replied:
Percy is no longer failing

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,753 @@
+
+## Native file system integration
+
+You may not immediately think of it, but the Podcasts app naturally integrates with the native file system. When I download a podcast episode, on my laptop it is stored in `~/Library/Group Containers/243LU875E5.groups.com.apple.podcasts/Library/Cache`. Unlike, say `~/Documents`, this directory is of course not meant to be accessed directly by regular users, but it is there.
```
```suggestion
You may not immediately think of it, but the Podcasts app naturally integrates with the native file system. When I download a podcast episode, on my laptop it is stored in `~/Library/Group Containers/243LU875E5.groups.com.apple.podcasts/Library/Cache`. Unlike, say `~/Documents`, this directory is of course not meant to be accessed directly by regular users, but it is there.
Other storage mechanisms than files are referenced in the [offline content](#offline-content-available-and-media-playable) section.
```

robdodson commented:
Percy changes LGTM'd. @kaycebasques should I assign this back to you?

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,754 @@
+
+## Multitasking and app shortcut
+
+Of course I can always multitask back to the Podcasts app from anywhere. The app has a clearly distinguishable icon that I can also put on my desktop or application dock so Podcasts can be launched immediately when I feel like it.
```
In this case I mean Podcasts the app name.

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,754 @@
+<div class="w-columns">
+  <figure class="w-figure">
+    <img src="./image2.png" alt="The Podcasts app in dark mode." width="350">
+    <figcaption class="w-figcaption">Podcasts supports light and dark mode.</figcaption>
```
```suggestion
    <figcaption class="w-figcaption">The Podcasts app supports light and dark mode.</figcaption>
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,754 @@
+  it should only be used for UI elements like button texts, etc.
+  The <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/font-family#<generic-name>:~:text=system%2Dui,-Glyphs"><code>system-ui</code></a>
+  value for the <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/font-family"><code>font-family</code></a> CSS property allows you to
+  specify the default UI font of the system to be used for your app, whatever it may be in the concrete case.
```
```suggestion
  specify the default UI font of the system to be used for your app.
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,754 @@
+  it should only be used for UI elements like button texts, etc.
+  The <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/font-family#<generic-name>:~:text=system%2Dui,-Glyphs"><code>system-ui</code></a>
+  value for the <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/font-family"><code>font-family</code></a> CSS property allows you to
+  specify the default UI font of the system to be used for your app, whatever it may be in the concrete case.
```
I just removed the last bit, this should make it clear.

tomayac commented:
@jpmedley Thanks for the last round of comments. Since you requested changes, you need to LGTM once more before it can be merged. PTAL. Thanks!

tomayac commented:
Thanks, @jpmedley! Looks like it's still waiting for either @samthor or @robdodson to LGTM. 

@kaycebasques, FYI, I will not burn this on a Friday, but merge on Tuesday (since Monday is a public holiday in large parts of the Christian world).

kaycebasques commented:
@tomayac Percy isn't failing anymore so you can merge this without a review from Sam or Rob when you're ready

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,754 @@
+---
+layout: post
+title: App-like Progressive Web Apps
```
```suggestion
title: Make your PWA feel more like an app
```

tomayac commented on src/site/content/en/progressive-web-apps/app-like-pwas/index.md:
```diff
@@ -0,0 +1,754 @@
+---
+layout: post
+title: App-like Progressive Web Apps
```
Done.

tomayac commented:
I'm sorry, but for some reason it's still requesting an LGTM from either @samthor or @robdodson. Could either of you quickly approve, please?
